### DISCUSSION TITLE
Question Regarding Arithmetic Operations and Raw SQL Queries in Prisma

### DISCUSSION THREAD

**Discussion by nrikiji** [prisma/prisma#21641] [Q&A] [answered]
### Question

Can you please explain how to express the following SQL query in Prisma?

```sql
select * from products where stock - reserved > 0
```

I have considered two approaches, but haven't been able to find a solution.

Approach 1:
Is it possible to use arithmetic operators between columns in Prisma?

Approach 2:
Is it possible to use rawQuery as part of the WHERE clause to achieve this?

Approach 3:
Can "Computed fields" be used for this kind of application?

I would appreciate any guidance or advice on how to achieve this with Prisma.

**Comment by ludralph** [comment, marked as answer]
Hi @nrikiji 👋 

Thank you for raising this question.

> Approach 1: Is it possible to use arithmetic operators between columns in Prisma?

Prisma currently does not support arithmetic operations between columns directly in the query. We do have a feature request for it [here](https://github.com/prisma/prisma/issues/5097).

> Approach 2: Is it possible to use rawQuery as part of the WHERE clause to achieve this?

This is also not supported at the moment. We do have a feature request for it [here](https://github.com/prisma/prisma/issues/5560)

>Approach 3: Can "Computed fields" be used for this kind of application?

Computed fields can be used for this kind of operation. There is a feature request for computed fields [here](https://github.com/prisma/prisma/issues/3394). However, you can also use the [Prisma Client Extension](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) for implementing computed fields. Here is an example of how you can achieve this.

Say you have a model defined below:
```prisma
model Product {
  id      Int    @id @default(autoincrement())
  name    String
  stock   Int
  reserved Int
```

You can write an extension for it as shown below
```js
const prisma = new PrismaClient().$extends({
  result: {
    product: {
      difference: {
        needs: { stock: true, reserved: true },
        compute(product) {
          return product.stock - product.reserved;
        },
      },
    },
  },
});
```

And use it as shown below:
```js
  const products = await prisma.product.findMany({});
  for (const product of products) {
    console.info(` ${product.difference}`);
  }
```

You can also achieve this using the `$queryRaw` method in Prisma.
```js
const result = await prisma.$queryRaw`SELECT * FROM products WHERE stock - reserved > 0`
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nrikiji** [reply]
Thank you for your answer! It looks like it's going to be resolved.
